src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createStatement = connection.prepareStatement(createQuery,
+                         Statement.RETURN_GENERATED_KEYS)) {
+
+            createStatement.setString(1, car.getModel());
+            createStatement.setLong(2, car.getManufacturer().getId());
+            createStatement.executeUpdate();
+            ResultSet resultSet = createStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create " + car, e);
+        }
+        insetrDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getQuery = "SELECT c.id as car_id, model, m.id as manufacturer_id, "
+                + "name as  manufacturer_name, country as manufacturer_country "
+                + "FROM cars c "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getCarStatement = connection.prepareStatement(getQuery)) {
+
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarWithManufacturer(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get driver by id " + id, e);
+        }
+        if (car != null) {
+            car.setDriver(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT c.id as car_id, model, m.id as manufacturer_id, "
+                + "name as  manufacturer_name, country as manufacturer_country "
+                + "FROM cars c "
+                + "JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllDriversStatement = connection
+                         .prepareStatement(getAllQuery)) {
+
+            ResultSet resultSet = getAllDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all ", e);
+        }
+        if (cars != null) {
+            for (Car car: cars) {
+                car.setDriver(getDriversForCar(car.getId()));
+            }
+        }
```

Comment:
Maybe would it be better to move this logic to another function

Reply:
I used stream api